R/hbv.R

```diff
@@ -0,0 +1,436 @@
+# HBV model code for hydromad
+# R and Rcpp code by Alexander Buzacott (abuz5257@uni.sydney.edu.au)
+#
+# Implementation based on HBV light as described in:
+# Seibert, J. and Vis, M. (2012). Teaching hydrological modeling with a user-
+# friendly catchment runoff-model software package. Hydrology and Earth System
+# Sciences, 16, 3315–3325, 2012.
+#
+#' @name hbv
+#' @md
+#' @aliases hbv.sim hbvrouting hbvrouting.sim
+#' @title HBV rainfall-runoff model
+#' @description An implementation of the HBV rainfall-runoff model.
+#' @param DATA A time-series like object with columns P (precipitation in mm),
+#' E (potential evapotranspiration in mm), T (average air temperature in ºC).
+#' @param tt Threshold temperature for snow and snow melt in degrees Celsius.
+#' @param cfmax Degree-day factor for snow melt (mm/(ºC.day)).
+#' @param sfcf Snowfall correction factor. Amount of precipitation below
+#' threshold temperature that should be rainfall instead of snow.
+#' @param cwh Liquid water holding capacity of the snowpack.
+#' @param cfr Refreezing coefficient for water in the snowpack.
+#' @param fc Maximum amount of soil moisture storage (mm).
+#' @param lp Threshold for reduction of evaporation. Limit for potential
+#' evapotranspiration.
+#' @param beta Shape coefficient in soil routine.
+#' @param return_state Whether to return the state variables.
+#'
+#' @param U Effective rainfall series
+#' @param k0 Recession coefficient (quick runoff).
+#' @param k1 Recession coefficient (upper groundwater storage).
+#' @param k2 Recession coefficient (lower groundwater storage).
+#' @param uzl Threshold for quick runoff for k0 outflow (mm).
+#' @param perc Maximum percolation from upper to lower groundwater storage.
+#' @param maxbas Routing, length of triangular weighting function.
+#' @param epsilon Values smaller than this in the output will be set to zero.
+#' @param return_components Whether to return state variables of the routing
+#' routine.
+#' @details This implementation of this HBV model closely follows the
+#' description of HBV light by Seibert and Vis, 2009. Daily average temperature
```

Comment:
Did you still want to do a check comparing with sample input/output from the HBVlight model?
We could add a disclaimer that direct compatibility is not guaranteed, but that might discourage people from using this implementation.

Reply:
Yep this is something I still want to confirm. I suspect it isn't exact at the moment due to the description of the triangular routing routine and accepting non-integer values. I'll try and have a look this weekend.

Reply:
Thanks! I am happy when you're happy - we just need to have a clear and convincing statement about what the status of this implementation is compared to other implementations of HBV.